src/testSession.ts

```diff
@@ -102,9 +111,14 @@ export class TestSession {
     // Write the test session options used to create this session
     fsCore.writeJsonSync(path.join(this.dir, 'testSessionOptions.json'), JSON.parse(JSON.stringify(options)));
 
+    // have to grab this before we change the home
+    transferExistingAuthToEnv();
+
     // Set the homedir used by this test, on the TestSession and the process
+    // TODO: does this work on Windows?
     process.env.HOME = this.homeDir = env.getString('TESTKIT_HOMEDIR', this.dir);
-
+    process.env.SFDX_USE_GENERIC_UNIX_KEYCHAIN = 'true';
```

Comment:
@mshanemc I was wondering if this was best left to the environment, rather that forcing to generic. Testkit home dir can be set by the test writer and if they choose to point home dir to their own home dir, then using generic keychain may be at odds with the keychain method used locally.

Reply:
On mac at least, home includes the keychain.  So if someone wants to reuse a hub, we want to use "their" keychain before testkit moves home.

When I didn't include this, you'd get a popup on the mac from keychain fussing about not having a keychain in the new home to store their stuff on.

I think that's a result of our os-dependent keychain stuff.  What **I don't know** is what happens when this runs on windows and whether we need additional logic around that.